src/experimental/tools/videoThumbnailLoader/load_array_buffer_data.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PPromise from "../../../utils/promise";
+
+/**
+ * Fetch data from URL
+ * @param {string} url
+ * @returns {Observable<ArrayBuffer>}
+ */
+export default function loadArrayBufferData(url: string): Promise<ArrayBuffer> {
```

Comment:
Why not using the `request` util?

```js
request({
  url,
  responseType: "arraybuffer",
  ignoreProgressEvents: true,
}).pipe(mergeMap(res => {
  // this is a response. Should even be type-checked by TypeScript
}));
```

Maybe `ignoreProgressEvents` should be `true` by default to simplify that API though.
I will work on it.